lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2391,6 +2391,17 @@ bool LLVMToSPIRV::transExecutionMode() {
           BM->addCapability(CapabilityFPGAKernelAttributesINTEL);
         }
       } break;
+
+      case spv::ExecutionModeDenormPreserve:
+      case spv::ExecutionModeDenormFlushToZero:
+      case spv::ExecutionModeSignedZeroInfNanPreserve:
+      case spv::ExecutionModeRoundingModeRTE:
+      case spv::ExecutionModeRoundingModeRTZ:{
+          unsigned TargetWidth;
```

Comment:
I think this should be guarded by an if about whether it is allowed to use corresponding extension or not.

By default, the translator is expected to not use any extensions and with this code, it won't be true. Moreover, with #537, this should trigger an assertion

Reply:
Fixed